lib/ui/home/home_screen.dart

```diff
@@ -112,8 +116,14 @@ class HomeScreenState extends ConsumerState<HomeScreen> {
       key: _scaffoldKey,
       backgroundColor: Colors.black,
       endDrawer: SideMenu(
-          sideMenuUIModel:
-              SideMenuUIModel(name: userProfile.email, version: version)),
+          sideMenuUIModel: SideMenuUIModel(
+            name: userProfile.email,
+            version: version,
+          ),
+          logoutCallback: () {
+            ref.read(homeViewModelProvider.notifier).logOut();
+            context.goNamed(RoutePath.login.name);
```

Comment:
```suggestion
            context.pushReplacementNamed(RoutePath.login.name);
```
I think in this screen we should remove all the routes before going back to the login screen to prevent users swipe the system's back button

Also, I think we should not call navigating to the login screen right after calling the logging out service. We should wait for the logging out call to return the success result and then call the navigation.

Reply:
Because logout is simply clear all storage so I don't think we need to wait for it as it return void

Reply:
<img width="800" alt="Screenshot 2023-05-19 at 1 46 15 PM" src="https://github.com/nimblehq/flutter-ic-david-bruce/assets/25218255/3c9d2662-e26c-41a4-93a5-dec60267a8c3">

There is an API of logging out 🤔 should we call it before clearng all storage